plugins/modules/packaging/os/pkgng.py

```diff
@@ -263,29 +265,46 @@ def install_packages(module, pkgng_path, packages, cached, pkgsite, dir_arg, sta
         if already_installed and state == "present":
             continue
 
-        update_available = query_update(module, pkgng_path, package, dir_arg, old_pkgng, pkgsite)
-        if not update_available and already_installed and state == "latest":
+        def update_available(): return query_update(module, pkgng_path, package, dir_arg, old_pkgng, pkgsite)
```

Comment:
Why not keep this as a variable?

Reply:
I think this is the least-refactoring way to make `update_available` lazily evaluate, only when the short-circuit boolean logic gets past the first `and`. The intent is to not call `query_update()` (thus avoiding an exec) if the module already knows that `already_installed == false` or is not targeting the `latest` state.

The prior behavior, when `update_available` was a simple variable, was that `query_update()` would be called for every package that was not already installed, wasting time on an exec of the `pkg` command with throw-away results.

Reply:
If you'd like to see it, I can definitely refactor the `if...continue` logic here to something that is more efficient in its exec calls and hopefully easier to read.

Reply:
There's a pep8 style issue to be fixed here, but I don't want to push that change and invalidate this review just yet.

ready_for_review

Reply:
I'm happy with the explanation. Please fix the pep8 fix :)

Reply:
Went back to fix the pep8 styling, and realized that your question meant that it would be hard for future maintainers to understand why it is an inline function and not a variable, so I refactored the `query_update()` call into the if expression. Keeps the performance improvement; less potential for confusion.